crates/anvil-polkadot/tests/it/snapshot.rs

```diff
@@ -0,0 +1,537 @@
+use anvil_polkadot::{
+    api_server::revive_conversions::{AlloyU256, ReviveAddress},
+    config::{AnvilNodeConfig, SubstrateNodeConfig},
+};
+use polkadot_sdk::pallet_revive::{
+    self,
+    evm::{Account, Block},
+};
+use subxt::utils::H160;
+
+async fn assert_block_number(node: &mut TestNode, n: u64) {
+    assert_eq!(std::convert::Into::<u64>::into(node.best_block_number().await), n);
+    tokio::time::sleep(Duration::from_millis(500)).await;
+    let best_block = unwrap_response::<Block>(
+        node.eth_rpc(EthRequest::EthGetBlockByNumber(alloy_eips::BlockNumberOrTag::Latest, false))
+            .await
+            .unwrap(),
+    )
+    .unwrap();
+    let n_as_u256 = polkadot_sdk::pallet_revive::U256::from(n);
+    assert_eq!(best_block.number, n_as_u256);
+
+    let finalized_block = unwrap_response::<Block>(
+        node.eth_rpc(EthRequest::EthGetBlockByNumber(
+            alloy_eips::BlockNumberOrTag::Finalized,
+            false,
+        ))
+        .await
+        .unwrap(),
+    )
+    .unwrap();
+    assert_eq!(finalized_block.number, n_as_u256);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_best_block_after_evm_revert() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config.clone(), substrate_node_config).await.unwrap();
+
+    // Assert on initial best block number.
+    assert_block_number(&mut node, 0).await;
+
+    // Snapshot at genesis.
+    let id = unwrap_response::<String>(node.eth_rpc(EthRequest::EvmSnapshot(())).await.unwrap())
+        .unwrap();
+    assert_eq!(id, "0x0".to_string());
+
+    // Mine 5 blocks and assert on the new best block.
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(5)), None)).await.unwrap())
+        .unwrap();
+    assert_block_number(&mut node, 5).await;
+
+    // Snapshot at block number 5.
+    let id = unwrap_response::<String>(node.eth_rpc(EthRequest::EvmSnapshot(())).await.unwrap())
+        .unwrap();
+    assert_eq!(id, "0x1".to_string());
+
+    // Mine 5 more blocks.
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(5)), None)).await.unwrap())
+        .unwrap();
+    assert_block_number(&mut node, 10).await;
+
+    // Snapshot again at block number 10.
+    let id = unwrap_response::<String>(node.eth_rpc(EthRequest::EvmSnapshot(())).await.unwrap())
+        .unwrap();
+    assert_eq!(id, "0x2".to_string());
+    assert_block_number(&mut node, 10).await;
+
+    // Mine 5 more blocks.
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(5)), None)).await.unwrap())
+        .unwrap();
+    assert_block_number(&mut node, 15).await;
+
+    // Revert to the second snapshot and assert best block number is 10.
+    let snapshot_id = U256::from_str_radix(id.trim_start_matches("0x"), 16).unwrap();
+    assert_eq!(snapshot_id, U256::from(2));
```

Comment:
why not 
`let snapshot_id = U256::from(2)`

Reply:
I think you meant this? [2b6fc22](https://github.com/paritytech/foundry-polkadot/pull/336/commits/2b6fc226617c363007b54eaca5b3433a7a7586e3)